bin/i18n/resources/apps/animations/sync_out.rb

```diff
@@ -35,27 +40,29 @@ def execute
               translations = JSON.load_file(i18n_spritelab_file_path)
               spritelab_manifest_builder.upload_localized_manifest(js_locale, translations)
             ensure
-              progress_bar.increment
+              mutex.synchronize do
+                @lang_progress_incr ||= (progress_bar.total - progress_bar.progress).to_f / pegasus_languages.size
+                progress_bar.progress += @lang_progress_incr
+              rescue ProgressBar::InvalidProgressError
+                progress_bar.finish
+              end
             end
+
+            progress_bar.finish
           end
 
           private
 
-          def pegasus_languages
-            @pegasus_languages ||= PegasusLanguages.get_crowdin_name_and_locale
+          def mutex
+            @mutex ||= Thread::Mutex.new
           end
```

Comment:
Can you give a bit more detail about what is mutex? It is the first time I see it.

Reply:
[`Thread::Mutex` implements a simple semaphore that can be used to coordinate access to shared data from multiple concurrent threads.](https://ruby-doc.org/core-3.1.1/Thread/Mutex.html)
`Thread::Mutex#synchronize` is used when we need to safely provide access to shared data to threads without the risk of causing a race condition